app/javascript/controllers/dynamic_form_field_controller.js

```diff
@@ -0,0 +1,15 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="dynamic-form-field"
+export default class extends Controller {
+  static targets = [ "field" ]
+  static values = {
+    index: String
```

Comment:
The index of this stimulus controller comes form the index from the `fields_for` helper.
https://guides.rubyonrails.org/form_helpers.html#the-fields-for-helper-index-option

It acts as a placeholder index for each of the new elements to add